src/diffusers/models/modeling_utils.py

```diff
@@ -671,10 +671,12 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             hf_quantizer = None
 
         if hf_quantizer is not None:
-            if device_map is not None:
+            is_bnb_quantization_method = hf_quantizer.quantization_config.quant_method.value == "bitsandbytes"
+            if is_bnb_quantization_method and device_map is not None:
```

Comment:
@sayakpaul @a-r-r-o-w  
I think we should handle this in validate_environment instead since:
1. it is specific to BNB
2. that's that method meant to do 
no?

Reply:
1. Yes, the line that was deleted, `if device_map is not None`, was added in the BnB quantization PR. I believe it might be because BnB does not support device_map when loading a pre-quantized model. [Here](https://github.com/huggingface/diffusers/pull/10009#discussion_r1861515390) is some additional discussion for context.

2. Yes, I think better to move in `validate_environment` if that would work. WDYT @sayakpaul? Would prefer to do in a separate PR though since we can keep this one more limited to torchao stuff that way